R/runEmployment.R

```diff
@@ -0,0 +1,304 @@
+#' Computes the employment values (jobs) across different sectors
+#' @param pathToMIF A mif file putput from a REMIND run
+#' @param improvements Either "None", "CEEW", "Dias", "Rutovitz_aus","Solar_found" or "All". Use "All" for all improvements.
+#' @param subtype Subtype for how shares of solar rooftop, wind offshore, and small hydro are assumed in the future. Options "current", "irena", and "expert". See calcDspvShare for more information.
+#' @param share_manf Either "current" or "local". Current implies current shares of world manufacture remain same until 2050, current means that in 2050 all countries manufacture required components locally.
+#' @param multiplier controls how the regional multiplier for non-oecd countries changes with time.
+#' @param decline How should the employment factors change over time? "capcosts" means according to capital costs. "static" means it doesn't change
+#' @description This function returns a magpie object containing the reporting the jobs for different technologies
+#' @return A magpie object
+#' @author Aman Malik
+#' @examples
+#' \dontrun{
+
+#' runEmployment(pathToMIF, improvements = "All", multiplier = "own", subtype = "expert", share_manf = "local", decline = "capcosts")
+#' }
+#' @importFrom madrat calcOutput
+#' @importFrom magclass getNames add_columns getItems
+#' @export
+
+reportEmployment <- function(pathToMIF, improvements, multiplier, subtype, share_manf, decline) {
+  
+  input_mif <- remind2::readReportingMIF(pathToMIF = "") # read as data frame
+  input_mif_mp <- as.magpie(input_mif) # convert to magpie object
+  input_mif_mp <- collapseDim(input_mif_mp) # remove dimensions with same values, i.e., "model" and "scenario"
+  input_mif_mp <- collapseDim(input_mif_mp,dim = 3.2) # remove "unit" dimension
+  
+  # use setConfig(forcecache=TRUE) to make the code run faster
+  # employment factors
+  x <- madrat::calcOutput("Employmentfactors", improvements = improvements, multiplier = multiplier)
+  getNames(x) <- gsub("Wind onshore","Wind|Onshore",getNames(x))
+  getNames(x) <- gsub("Wind offshore","Wind|Offshore",getNames(x))
+  x <- x[, , "HP", pmatch = TRUE, invert = TRUE] # excluding (at the moment) jobs in combined heat and power plants
+  
+  ## DECLINE FACTORS------------------------
+  if (decline == "capcosts") {
+    
+
+    # capital costs and OM fixed costs evolution over time for different techs, used to calculated the decline factor
+   # report_tech <- reportTechnology(gdx)
+    #report_tech <- collapseNames(report_tech)
```

Comment:
Can you explain in a comment why these lines are commented out and when/under which circumstances they should be commented in again? If not, maybe it's better to delete them for now?